ksqldb-engine/src/main/java/io/confluent/ksql/serde/InternalFormats.java

```diff
@@ -50,10 +52,15 @@ private InternalFormats() {
    * @see SerdeFeaturesFactory#buildInternal
    */
   public static Formats of(final KeyFormat keyFormat, final ValueFormat valueFormat) {
+    // Do not use NONE format for internal topics:
```

Comment:
Is the thinking here that JSON works even if no key columns are specified, so we might as well always use JSON so that if a stream with no key columns is the source for a another stream/table down the line, then the sink stream/table will automatically get JSON as the key format, regardless of what the config default key format is?

Reply:
Perhaps I've missed it but I don't see any QTT for testing this. Can we add one in order to prevent regressions here?

Reply:
>  if no key columns are specified

Internal topics _always_ have a key column.

This is just stop-gap functionality. I'm working on better handling of internal key format in another PR.

Reply:
If that's the case, why would we ever arrive in this situation (of trying to build an internal topic with key format NONE)? Can we just fail if that happens, instead of converting the format to JSON?